### ISSUE TITLE
AttributeManager.setAttribute is megamorphic

### ISSUE THREAD

**Issue by chadhietala** [tildeio/glimmer#338]

**Comment by stefanpenner** [comment]
@chadhietala do you mean this one https://github.com/tildeio/glimmer/blob/8409f1e8b6eae6ca323aea45ad0d69ab9d05e586/packages/glimmer-runtime/lib/dom/attribute-managers.ts#L72 ? Im not sure how to make it not megamorphic, as it will interact with a large number of different input types.

What did you have in mind?

**Comment by chadhietala** [comment]
I agree. @krisselden mentioned this to me the other day on slack. I more or less created this issue because when Kris pings me with things I tend to believe they should be investigated 😄 .

**Comment by locks** [comment]
Is this still relevant?
